src/redux/slice/sliceCart.js

```diff
@@ -27,24 +27,24 @@ const orderSlice = createSlice({
   initialState,
   reducers: {
     increaseProduct(state, action) {
-      const itemIndex = state.cartItems.findIndex(cartElement => cartElement.id === action.payload.id);
+      const itemIndex = state.cartItems.findIndex(cartElement => cartElement.id === action.payload.id && cartElement.color === action.payload.color && cartElement.size === action.payload.size);
       if (itemIndex >= 0) {
         state.cartItems[itemIndex].cartQuantity += 1;
       } else {
-        const conProduct = { ...action.payload, cartQuantity: 1 };
-        state.cartItems.push(conProduct);
+        const product = { ...action.payload, cartQuantity: 1 };
+        state.cartItems.push(product);
```

Comment:
One of the key rules concerning reducers is that they _must not mutate state_. 

This means that we should never modify the `state` passed to the reducer like this: 
`state.cartItems.push(product);` ❌ 
or:
`state.cartItems = nextCartItems;` ❌

Instead, we should create a new object using state and action. A great way to do it is to use destructuring:
```
const updatedCartItems = [
   ...state.cartItems, 
   product
];
prepareCartTotal({
   ...state,
   cartItems: updatedCartItems,
});
```
and: in `decreaseProduct` reducer:
```
{
   ...state,
   cartItems: nextCartItems,
};
```
More info about rules of the reducers can be found in this very helpful guide: https://redux.js.org/tutorials/fundamentals/part-3-state-actions-reducers#rules-of-reducers